build_tools/benchmarks/post_benchmark_comment.py

```diff
@@ -173,17 +191,45 @@ def main(args: argparse.Namespace):
   if github_token is None:
     raise ValueError("GITHUB_TOKEN must be set.")
 
-  gist_bot_user = os.environ.get("GIST_BOT_USER")
-  if gist_bot_user is None:
-    raise ValueError("GIST_BOT_USER must be set.")
+  comment_bot_user = os.environ.get("COMMENT_BOT_USER")
+  if comment_bot_user is None:
+    raise ValueError("COMMENT_BOT_USER must be set.")
 
   gist_bot_token = os.environ.get("GIST_BOT_TOKEN")
   if gist_bot_token is None:
     raise ValueError("GIST_BOT_TOKEN must be set.")
 
-  pr_number = args.pr_number
   comment_data = benchmark_comment.CommentData(
       **json.loads(args.comment_json.read_text()))
+  # Sanitize the pr number to make sure it is an integer.
+  pr_number = int(comment_data.unverified_pr_number)
+
+  if args.no_verify_pr:
+    github_event = None
+  else:
+    github_event = json.loads(args.github_event_json.read_text())
+
+  pr_client = GithubClient(requester=APIRequester(github_token=github_token))
+  if github_event is not None:
+    workflow_run_sha = github_event["workflow_run"]["head_sha"]
+    pr_head_sha = pr_client.get_pull_request_head_commit(pr_number=pr_number)
+    # We can't get the trusted PR number of a workflow run from GitHub API. So we
+    # take the untrusted PR number from presubmit workflow and verify if the PR's
+    # current head SHA matches the commit SHA in the workflow run. It assumes that
+    # to generate the malicious comment data, attacker must modify the code and
+    # has a new commit SHA. So if the PR head commit matches the workflow run with
+    # attacker's commit, either the PR is created by the attacker or other's PR
+    # has the malicious commit. In both cases posting malicious comment is
+    # acceptable.
+    #
+    # Note that the collision of a target SHA1 is possible but GitHub has some
+    # protections (https://github.blog/2017-03-20-sha-1-collision-detection-on-github-com/).
+    # The assumption also only holds if files in GCS can't be overwritten (so the
+    # comment data can't be modified without changing the code).
```

Comment:
That is currently how we have permissions configured: presubmit service account has write but not delete permissions, so cannot overwrite